### ISSUE TITLE
Comments

### ISSUE THREAD

**Issue by mattbk** [hmngwy/weblog.sh#3]
Probably not a priority, but commenting would be a useful feature.  Or would that make it less minimal? ;-)

**Comment by hmngwy** [comment]
Yeah I was thinking about that. I don't think it would impact minimalism, I think a commenting system is separate from the blogging system, in the sense that it largely caters to a different user group.

How do you picture the comment system?

**Comment by jaapz** [comment]
I think the best way to handle this would just be to add a disqus comment component

**Comment by GSI** [comment]
Let me ask a stupid question: What's your use case for comments? :)

**Comment by hmngwy** [comment]
@GSI I'm not sure what you mean, could you elaborate?

**Comment by jedahan** [comment]
The most minimal thing I could think of - add a new directory called comments. Any POST to a current article would create a new comment in that folder, maybe with the datetime. Allow blog writers to decide how they want to interact with that. For me, I would probably cronjob to do something indieweb-style: http://indiewebify.me/ and http://aaronparecki.com/articles/2013/10/13/1/realtime-indieweb-comments . Since we have markdown, individuals can decide to add forms directly in the page if they want using html, with whatever fields they want. Fields could be liquid templates style in the header, body as body.

-1 on disqus or reimplementing a weblog.sh version of the very same javascripty thing.

**Comment by hmngwy** [comment]
@jedahan @jaapz 

For the most part, what worries me is that comments are usually spam magnets, at the current server setup we can't afford any more additional services/daemons to employ spam detection.

Re disqus, it loads javascript. :(

Re forms through markdown, html will be disabled in markdown support soon, any html will be stripped. 

Edit: If there will be commenting support, when we can afford it, it will likely be supported inline (like a weblog.sh supplied form).

**Comment by jedahan** [comment]
OK, how about more minimal: instead of posting the contents on the server, just append the url contained in a webmention to walt@weblog.sh:webmentions/blog-post-title.md .

The higher-level question I have is: how can I allow for someone to tell me 'hey I have published a reply/want to participate in a discussion about this' without it tying back to an email address or third-party service.

**Comment by hmngwy** [comment]
@jedahan

> OK, how about more minimal: instead of posting the contents on the server, just append the url contained in a webmention to walt@weblog.sh:webmentions/blog-post-title.md .

Have not read about webmentions before, I just read the [spec](https://github.com/converspace/webmention). Seems like lighter version of pingback. 

Like any spec, the commenter will have to be on a platform that supports webmentions, a bit limiting atm  don't you think? This idea is valid though, it doesn't force weblog.sh to apply anything that another inline discussion solution wouldn't (backend moderation, spam detection). Also, I'm a fan of the idea "if you want to talk about it, then write about it on your blog", I forgot who said that.

> The higher-level question I have is: how can I allow for someone to tell me 'hey I have published a reply/want to participate in a discussion about this' without it tying back to an email address or third-party service.

I don't think it's possible to have push notifications, since weblog.sh will never know how to contact its users. There will likely be a backend to moderate comments/webmentions.

**Comment by pscollins** [comment]
> I don't think it's possible to have push notifications, since weblog.sh will never know how to contact its users.

One thought --- maybe a bit hacky --- keep around a "notifications.txt" file in each user's home directory and append to the end of it whenever something interesting happens to them. Let them deal with scp'ing it down and clearing it out every once in a while. 

Not exactly push notifications, but it seems to be in line with the spirit of the rest of the design.

**Comment by jaapz** [comment]
> Re disqus, it loads javascript. :(

Is that against what weblog.sh stands for? I don't think it's inherently bad to load some javascript if you get such an easy comment system basically for free.

**Comment by GSI** [comment]
@hmngwy My question is based on my personal (and limited) experience with comments in blogs:
- Why should I care if the blogger/article is in accordance with my opinion?
- If there's something really worth stating in regards to an article I'd write an email to the author directly.
- If a point is important enough to make, I totally agree with your quote "if you want to talk about it, then write about it on your blog"
- If I want to discuss (I don't), I can go to another place like Reddit or some forums.

There are cases where I draw value from comments, but I could live without them just fine.

For commercial sites I understand that comments even mean money.

This is how I did comments back when I still used them:

https://caurea.org/2012/03/31/this-blog-has-comments-again.html

(I'd go with the same approach again, if the need arised.)

Apart from that, webmentions, even if not widely adopted yet, would be my preference.

@jaapz Allowing authors to publish anonymously while having visitors be tracked all across the internet via a service like Disqus actually does appear inherently bad to me :) Also, most authors are likely to want to have a look at their final piece of work, so any (external) JS would compromise anonymity.

**Comment by mattbk** [comment]
@hmngwy I don't think it needs to be global. I was thinking of the same sort of @-mention like @jedahan suggested, except that you could limit the backtracing to weblog.sh.

**Comment by hmngwy** [comment]
@jaapz exactly what @GSI said. It is really tempting because it's an easy solution, but it sacrifices other aspects of the service, yes some users can just disable javascript, but I prefer weblog.sh to have an active stance in keeping users as anonymous as possible.

@GSI

Re: the points you listed > Same. Personally, I don't enable comments on my blogs. Almost 80% of the blogs I frequently visit don't have comments though, so my opinion is also very limited. I do see the value in it, this thread for example. So if some users want comments, they should be able to opt in.

I like the solution you described in your blog, no need for forms, same backend requirements as any other inline solution and possibly just use built-in spam detection software applied by the email server. So far this seems to be the simplest.

**Comment by cserb** [comment]
@hmngwy how about a way to answer to a blog post by posting another post. A bit like Twitter.
You can make reference like @~postmaster/title inside the blog post? You can link it internally as a comment and display it under a blogpost either showing the titles of the comments in a list directly or you have to click to see a list of all comments?

**Comment by jedahan** [comment]
@GSI that is a pretty awesome solution, that currently requires no change on weblog.sh's end. I like how opening an email client might affect how a commenter decides to compose their message. The only downside I can see is that it ties the user back to an email address, which might matter to some.

I still think I prefer webmention to a file called mentions.txt in your home directory: If you want to comment on this blog, post your comment anywhere publically accessible, webmention the address. It could be another weblog.sh url, it could be a tweet, it could be pastebin, etc.

It would be a fun day-project to create a pastebin + webmention third party commenting system, if people want to use that...

**Comment by hmngwy** [comment]
@cserb hmm.. that would close the commenting to users of the system only, what we want to aim for is something that would allow anyone to participate in the discussion.

**Comment by jaapz** [comment]
@GSI @hmngwy the privacy thing I hadn't thought about, and that's a pretty good point. I am partial to the e-mail-comment idea now. It raises the bar for commenters which can be either a good or bad thing. But then there should be a way to make sure that the commenter's e-mailadres is discarded the moment the comment is processed.

**Comment by githubgnulinux** [comment]
I plan to use the comment function of diaspora; post to weblog, post weblog link to diaspora and people may comment on the diaspora system. Too much ecmascript is used by diaspora, but it's OK, especially when images are disabled.
